lib/plugins/scale.js

```diff
@@ -0,0 +1,22 @@
+export default shower => {
+    const { container } = shower;
+    const resizeObserver = new ResizeObserver(entries => {
```

Comment:
I've managed to remove all `document` references except one (querying container element).
That makes Shower more portable with little to no cost.

`ResizeObserver` ships in Chrome 64. It has an API that is very unlikely to change and allows us to have that portability without tying to element structure. However, it is not a web standard yet.

@pepelsbey, need you insight on this & other comments (some of them are marked as "outdated", but hey are not).

Reply:
I think `ResizeObserver` is here to stay, one way or another. We’re living in observers trend these days :) How about _Latest stable versions of Chrome, Internet Explorer, Firefox, Opera and Safari_ compatibility promise?

Reply:
I've added a [polyfill](https://github.com/que-etc/resize-observer-polyfill) in https://github.com/shower/core/pull/134/commits/a60e8f69c7bc85bfc56ba18a6209c03cfb43f66a: it has good code & precise observation strategy.